File: 백준 문제 풀이 모음/class 2/10773.py

class stack() :
    def __init__(self) :
        self.stack = []

    def push(self, data) :
        self.stack.append(data)
    
    def pop(self) :
        self.stack.pop()
    
    def sum(self) :
        return(sum(self.stack))
    
import sys
num = int(sys.stdin.readline())
stk = stack()

for i in range(num) :
    x = int(sys.stdin.readline())

    if x == 0 :
        stk.pop()
    
    else :
        stk.push(x)


print(stk.sum())

File: 백준 문제 풀이 모음/kotlin/class 2/10773.kt
import java.util.*

fun main () = with(System.`in`.bufferedReader()) {
    val stack = Stack<Int>()

    val n = readLine().toInt()
    repeat(n) {
        val input = readLine().toInt()
        if (0 == input) {
            stack.pop()
        } else {
            stack.add(input)
        }
    }

    println(stack.sum())




}